mqtt/mqttd/src/time.rs

```diff
@@ -0,0 +1,57 @@
+//! `tokio::time::delay_for` and `tokio::time::delay_until` has a bug that
```

Comment:
why not intro a simple while loop like that
```
while renew_at > 0 
do
 interval = min(renew_at, 30d)
 delay_for(interval)
 renew_at -= interval
end

Reply:
Yes, we could do that. I thought we may need more general solution

Reply:
Yes, we have similar logic in EdgeHub as well - https://github.com/Azure/iotedge/blob/master/edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Service/CertificateRenewal.cs

Reply:
it is coupled with certificates renewal, and we may need more general solution for "long sleep" action. 

since async/await is a generated by the compiler the type will be harder to understand when awaiting sleep and something else at the same time. 
in the end we have the same logic as @vadim-kovalyov mentioned but written with future rather than async method.